File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/actions/fight/GameActionMarkedCell.scala





















// Generated on 05/03/2013 10:35:02
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait GameActionMarkedCell extends TypeDeserializer {
  val typeId = 85

  protected trait Type extends BaseType with Serializable {
    val typeId = GameActionMarkedCell.typeId

    val cellId: Short
    val zoneSize: Byte
    val cellColor: Int
    val cellsType: Byte

    def serialize(buf: Buffer) {
      buf.writeShort(cellId)
      buf.writeByte(zoneSize)
      buf.writeInt(cellColor)
      buf.writeByte(cellsType)
    }

  }

  protected def create(cellId: Short, zoneSize: Byte, cellColor: Int, cellsType: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val cellId = buf.readShort()
    val zoneSize = buf.readByte()
    val cellColor = buf.readInt()
    val cellsType = buf.readByte()

    create(cellId, zoneSize, cellColor, cellsType)
  }

}

object GameActionMarkedCell extends GameActionMarkedCell {
  case class TypeImpl(cellId: Short, zoneSize: Byte, cellColor: Int, cellsType: Byte) extends Type
  override def create(cellId: Short, zoneSize: Byte, cellColor: Int, cellsType: Byte) = TypeImpl(cellId, zoneSize, cellColor, cellsType)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(cellId: Short, zoneSize: Byte, cellColor: Int, cellsType: Byte) = create(cellId, zoneSize, cellColor, cellsType)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/GameActionMarkedCell.java


// Generated on 05/08/2013 19:38:00
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameActionMarkedCell extends NetworkType {
    public static final short TYPE_ID = 85;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short cellId;
    public byte zoneSize;
    public int cellColor;
    public byte cellsType;
    
    public GameActionMarkedCell() { }
    
    public GameActionMarkedCell(short cellId, byte zoneSize, int cellColor, byte cellsType) {
        this.cellId = cellId;
        this.zoneSize = zoneSize;
        this.cellColor = cellColor;
        this.cellsType = cellsType;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(cellId);
        buf.writeByte(zoneSize);
        buf.writeInt(cellColor);
        buf.writeByte(cellsType);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        cellId = buf.readShort();
        if (cellId < 0 || cellId > 559)
            throw new RuntimeException("Forbidden value on cellId = " + cellId + ", it doesn't respect the following condition : cellId < 0 || cellId > 559");
        zoneSize = buf.readByte();
        cellColor = buf.readInt();
        cellsType = buf.readByte();
    }
    
}
